<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Agglomerate or merge data using taxonomic information — agglomerate-methods • mia</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Agglomerate or merge data using taxonomic information — agglomerate-methods"><meta property="og:description" content="Agglomeration functions can be used to sum-up data based on specific criteria
such as taxonomic ranks, variables or prevalence.
agglomerateByRanks takes a SummarizedExperiment, splits it along the
taxonomic ranks, aggregates the data per rank, converts the input to a
SingleCellExperiment objects and stores the aggregated data as
alternative experiments. unsplitByRanks takes these alternative
experiments and flattens them again into a single
SummarizedExperiment."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mia</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.34</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/mia.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/microbiome/mia/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Agglomerate or merge data using taxonomic information</h1>
    <small class="dont-index">Source: <a href="https://github.com/microbiome/mia/blob/devel/R/agglomerate.R" class="external-link"><code>R/agglomerate.R</code></a>, <a href="https://github.com/microbiome/mia/blob/devel/R/splitByRanks.R" class="external-link"><code>R/splitByRanks.R</code></a></small>
    <div class="hidden name"><code>agglomerate-methods.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Agglomeration functions can be used to sum-up data based on specific criteria
such as taxonomic ranks, variables or prevalence.</p>
<p><code>agglomerateByRanks</code> takes a <code>SummarizedExperiment</code>, splits it along the
taxonomic ranks, aggregates the data per rank, converts the input to a
<code>SingleCellExperiment</code> objects and stores the aggregated data as
alternative experiments. <code>unsplitByRanks</code> takes these alternative
experiments and flattens them again into a single
<code>SummarizedExperiment</code>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">agglomerateByRank</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">agglomerateByVariable</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">agglomerateByRank</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  rank <span class="op">=</span> <span class="fu"><a href="taxonomy-methods.html">taxonomyRanks</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  empty.fields <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="st">""</span>, <span class="st">" "</span>, <span class="st">"\t"</span>, <span class="st">"-"</span>, <span class="st">"_"</span><span class="op">)</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">agglomerateByVariable</span><span class="op">(</span><span class="va">x</span>, <span class="va">by</span>, <span class="va">f</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'TreeSummarizedExperiment'</span></span>
<span><span class="fu">agglomerateByVariable</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">by</span>,</span>
<span>  <span class="va">f</span>,</span>
<span>  update.tree <span class="op">=</span> <span class="va">mergeTree</span>,</span>
<span>  mergeTree <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SingleCellExperiment'</span></span>
<span><span class="fu">agglomerateByRank</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">...</span>,</span>
<span>  altexp <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  altexp.rm <span class="op">=</span> <span class="va">strip_altexp</span>,</span>
<span>  strip_altexp <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'TreeSummarizedExperiment'</span></span>
<span><span class="fu">agglomerateByRank</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">...</span>,</span>
<span>  update.tree <span class="op">=</span> <span class="va">agglomerateTree</span>,</span>
<span>  agglomerate.tree <span class="op">=</span> <span class="va">agglomerateTree</span>,</span>
<span>  agglomerateTree <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">agglomerateByRanks</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">agglomerateByRanks</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  ranks <span class="op">=</span> <span class="fu"><a href="taxonomy-methods.html">taxonomyRanks</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  as.list <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SingleCellExperiment'</span></span>
<span><span class="fu">agglomerateByRanks</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  ranks <span class="op">=</span> <span class="fu"><a href="taxonomy-methods.html">taxonomyRanks</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  as.list <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'TreeSummarizedExperiment'</span></span>
<span><span class="fu">agglomerateByRanks</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  ranks <span class="op">=</span> <span class="fu"><a href="taxonomy-methods.html">taxonomyRanks</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  as.list <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">splitByRanks</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">unsplitByRanks</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SingleCellExperiment'</span></span>
<span><span class="fu">unsplitByRanks</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  ranks <span class="op">=</span> <span class="fu"><a href="taxonomy-methods.html">taxonomyRanks</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,</span>
<span>  keep.dimred <span class="op">=</span> <span class="va">keep_reducedDims</span>,</span>
<span>  keep_reducedDims <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'TreeSummarizedExperiment'</span></span>
<span><span class="fu">unsplitByRanks</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  ranks <span class="op">=</span> <span class="fu"><a href="taxonomy-methods.html">taxonomyRanks</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,</span>
<span>  keep.dimred <span class="op">=</span> <span class="va">keep_reducedDims</span>,</span>
<span>  keep_reducedDims <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>a numeric matrix with samples as rows or a
<code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></code>
object.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>arguments passed to <code>agglomerateByRank</code> function for
<code>SummarizedExperiment</code> objects and other functions.
See <code>agglomerateByRank</code> for more details.</p></dd>


<dt id="arg-rank">rank<a class="anchor" aria-label="anchor" href="#arg-rank"></a></dt>
<dd><p><code>Character scalar</code>. Defines a taxonomic rank. Must be a value of
<code><a href="taxonomy-methods.html">taxonomyRanks()</a></code> function.</p></dd>


<dt id="arg-na-rm">na.rm<a class="anchor" aria-label="anchor" href="#arg-na-rm"></a></dt>
<dd><p><code>Logical scalar</code>. Should NA values be omitted when calculating
prevalence? (Default: <code>TRUE</code>)</p></dd>


<dt id="arg-empty-fields">empty.fields<a class="anchor" aria-label="anchor" href="#arg-empty-fields"></a></dt>
<dd><p><code>Character vector</code>. Defines which values should be
regarded as empty. (Default: <code>c(NA, "", " ", "\t")</code>). They will be
removed if <code>na.rm = TRUE</code> before agglomeration.</p></dd>


<dt id="arg-by">by<a class="anchor" aria-label="anchor" href="#arg-by"></a></dt>
<dd><p><code>Character scalar</code>. Determines if data is merged
row-wise / for features ('rows') or column-wise / for samples ('cols').
Must be <code>'rows'</code> or <code>'cols'</code>.</p></dd>


<dt id="arg-f">f<a class="anchor" aria-label="anchor" href="#arg-f"></a></dt>
<dd><p>A factor for merging. Must be the same length as
<code>nrow(x)/ncol(x)</code>. Rows/Cols corresponding to the same level will be
merged. If <code>length(levels(f)) == nrow(x)/ncol(x)</code>, <code>x</code> will be
returned unchanged.</p></dd>


<dt id="arg-update-tree">update.tree<a class="anchor" aria-label="anchor" href="#arg-update-tree"></a></dt>
<dd><p><code>Logical scalar</code>. Should
<code>rowTree()</code> also be merged? (Default: <code>FALSE</code>)</p></dd>


<dt id="arg-mergetree">mergeTree<a class="anchor" aria-label="anchor" href="#arg-mergetree"></a></dt>
<dd><p>Deprecated. Use <code>update.tree</code> instead.</p></dd>


<dt id="arg-altexp">altexp<a class="anchor" aria-label="anchor" href="#arg-altexp"></a></dt>
<dd><p><code>Character scalar</code> or <code>integer scalar</code>.
Specifies an alternative experiment containing the input data.</p></dd>


<dt id="arg-altexp-rm">altexp.rm<a class="anchor" aria-label="anchor" href="#arg-altexp-rm"></a></dt>
<dd><p><code>Logical scalar</code>. Should alternative
experiments be removed prior to agglomeration? This prevents too many
nested alternative experiments by default. (Default:
<code>TRUE</code>)</p></dd>


<dt id="arg-strip-altexp">strip_altexp<a class="anchor" aria-label="anchor" href="#arg-strip-altexp"></a></dt>
<dd><p>Deprecated. Use <code>altexp.rm</code> instead.</p></dd>


<dt id="arg-agglomerate-tree">agglomerate.tree<a class="anchor" aria-label="anchor" href="#arg-agglomerate-tree"></a></dt>
<dd><p>Deprecated. Use <code>update.tree</code> instead.</p></dd>


<dt id="arg-agglomeratetree">agglomerateTree<a class="anchor" aria-label="anchor" href="#arg-agglomeratetree"></a></dt>
<dd><p>Deprecated. Use <code>update.tree</code> instead.</p></dd>


<dt id="arg-ranks">ranks<a class="anchor" aria-label="anchor" href="#arg-ranks"></a></dt>
<dd><p><code>Character vector</code>. Defines taxonomic ranks. Must all be values
of <code><a href="taxonomy-methods.html">taxonomyRanks()</a></code> function.</p></dd>


<dt id="arg-as-list">as.list<a class="anchor" aria-label="anchor" href="#arg-as-list"></a></dt>
<dd><p><code>Logical scalar</code>. Should the list of
<code>SummarizedExperiment</code> objects be returned by the function
<code>agglomerateByRanks</code> as a SimpleList or stored in altExps?
(Default: <code>FALSE</code>)</p></dd>


<dt id="arg-keep-dimred">keep.dimred<a class="anchor" aria-label="anchor" href="#arg-keep-dimred"></a></dt>
<dd><p><code>Logical scalar</code>. Should the
<code>reducedDims(x)</code> be transferred to the result? Please note, that this
breaks the link between the data used to calculate the reduced dims.
(Default: <code>FALSE</code>)</p></dd>


<dt id="arg-keep-reduceddims">keep_reducedDims<a class="anchor" aria-label="anchor" href="#arg-keep-reduceddims"></a></dt>
<dd><p>Deprecated. Use <code>keep.dimred</code> instead.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p><code>agglomerateByRank</code> returns a taxonomically-agglomerated,
optionally-pruned object of the same class as <code>x</code>.
<code>agglomerateByVariable</code> returns an object of the same class as <code>x</code>
with the specified entries merged into one entry in all relevant components.
<code>agglomerateByRank</code> returns a taxonomically-agglomerated,
optionally-pruned object of the same class as <code>x</code>.</p>
<p>For <code>agglomerateByRanks</code>:
If <code>as.list = TRUE</code> : <code>SummarizedExperiment</code> objects in a
<code>SimpleList</code>
If <code>as.list = FALSE</code> : The <code>SummarizedExperiment</code> passed as a
parameter and now containing the <code>SummarizedExperiment</code> objects in its
altExps</p>
<p>For <code>unsplitByRanks</code>: <code>x</code>, with <code>rowData</code> and <code>assay</code>
data replaced by the unsplit data. <code>colData</code> of x is kept as well
and any existing <code>rowTree</code> is dropped as well, since existing
<code>rowLinks</code> are not valid anymore.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p><code>agglomerateByRank</code> can be used to sum up data based on associations
with certain taxonomic ranks, as defined in <code>rowData</code>. Only available
<code><a href="taxonomy-methods.html">taxonomyRanks</a></code> can be used.</p>
<p><code>agglomerateByVariable</code> merges data on rows or columns of a
<code>SummarizedExperiment</code> as defined by a <code>factor</code> alongside the
chosen dimension. This function allows agglomeration of data based on other
variables than taxonomy ranks.
Metadata from the <code>rowData</code> or <code>colData</code> are
retained as defined by <code>archetype</code>.
<code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></code> are
agglomerated, i.e. summed up. If the assay contains values other than counts
or absolute values, this can lead to meaningless values being produced.</p>
<p>Agglomeration sums up the values of assays at the specified taxonomic level. With
certain assays, e.g. those that include binary or negative values, this summing
can produce meaningless values. In those cases, consider performing agglomeration
first, and then applying the transformation afterwards.</p>
<p><code>agglomerateByVariable</code> works similarly to
<code><a href="https://rdrr.io/pkg/scuttle/man/sumCountsAcrossFeatures.html" class="external-link">sumCountsAcrossFeatures</a></code>.
However, additional support for <code>TreeSummarizedExperiment</code> was added and
science field agnostic names were used. In addition the <code>archetype</code>
argument lets the user select how to preserve row or column data.</p>
<p>For merge data of assays the function from <code>scuttle</code> are used.</p>
<p><code>agglomerateByRanks</code> will use by default all available taxonomic ranks, but
this can be controlled by setting <code>ranks</code> manually. <code>NA</code> values
are removed by default, since they would not make sense, if the result
should be used for <code>unsplitByRanks</code> at some point. The input data
remains unchanged in the returned <code>SingleCellExperiment</code> objects.</p>
<p><code>unsplitByRanks</code> will remove any <code>NA</code> value on each taxonomic rank
so that no ambiguous data is created. In additional, a column
<code>taxonomicLevel</code> is created or overwritten in the <code>rowData</code> to
specify from which alternative experiment this originates from. This can also
be used for <code><a href="https://rdrr.io/pkg/SingleCellExperiment/man/splitAltExps.html" class="external-link">splitAltExps</a></code> to
split the result along the same factor again. The input data from the base
objects is not returned, only the data from the <code>altExp()</code>. Be aware that
changes to <code>rowData</code> of the base object are not returned, whereas only
the <code>colData</code> of the base object is kept.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="splitOn.html">splitOn</a></code>
<code><a href="splitOn.html">unsplitOn</a></code>
<code>agglomerateByVariable</code>,
<code><a href="https://rdrr.io/pkg/scuttle/man/sumCountsAcrossFeatures.html" class="external-link">sumCountsAcrossFeatures</a></code>,
<code>agglomerateByRank</code>,
<code><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExps</a></code>,
<code><a href="https://rdrr.io/pkg/SingleCellExperiment/man/splitAltExps.html" class="external-link">splitAltExps</a></code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### Agglomerate data based on taxonomic information</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># print the available taxonomic ranks</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Kingdom" "Phylum"  "Class"   "Order"   "Family"  "Genus"   "Species"</span>
<span class="r-in"><span><span class="fu"><a href="taxonomy-methods.html">taxonomyRanks</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Kingdom" "Phylum"  "Class"   "Order"   "Family"  "Genus"   "Species"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># agglomerate at the Family taxonomic rank</span></span></span>
<span class="r-in"><span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu">agglomerateByRank</span><span class="op">(</span><span class="va">GlobalPatterns</span>, rank<span class="op">=</span><span class="st">"Family"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">## How many taxa before/after agglomeration?</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 19216</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">x1</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 341</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># agglomerate the tree as well</span></span></span>
<span class="r-in"><span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="fu">agglomerateByRank</span><span class="op">(</span><span class="va">GlobalPatterns</span>, rank<span class="op">=</span><span class="st">"Family"</span>,</span></span>
<span class="r-in"><span>                       update.tree <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">x2</span><span class="op">)</span> <span class="co"># same number of rows, but</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 341</span>
<span class="r-in"><span><span class="fu">rowTree</span><span class="op">(</span><span class="va">x1</span><span class="op">)</span> <span class="co"># ... different</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Phylogenetic tree with 19216 tips and 19215 internal nodes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Tip labels:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   549322, 522457, 951, 244423, 586076, 246140, ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Node labels:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   , 0.858.4, 1.000.154, 0.764.3, 0.995.2, 1.000.2, ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Rooted; includes branch lengths.</span>
<span class="r-in"><span><span class="fu">rowTree</span><span class="op">(</span><span class="va">x2</span><span class="op">)</span> <span class="co"># ... tree</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Phylogenetic tree with 341 tips and 340 internal nodes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Tip labels:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   951, 215972, 138353, 546313, 173903, 202347, ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Node labels:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   , 0.858.4, 0.764.3, 0.985.6, 1.000.112, 0.978.18, ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Rooted; includes branch lengths.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># If assay contains binary or negative values, summing might lead to</span></span></span>
<span class="r-in"><span><span class="co"># meaningless values, and you will get a warning. In these cases, you might</span></span></span>
<span class="r-in"><span><span class="co"># want to do agglomeration again at chosen taxonomic level.</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="transformAssay.html">transformAssay</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, method <span class="op">=</span> <span class="st">"pa"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">agglomerateByRank</span><span class="op">(</span><span class="va">tse</span>, rank <span class="op">=</span> <span class="st">"Genus"</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'pa' includes binary values.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> Agglomeration of it might lead to meaningless values.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> Check the assay, and consider doing transformation againmanually with agglomerated data.</span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="transformAssay.html">transformAssay</a></span><span class="op">(</span><span class="va">tse</span>, method <span class="op">=</span> <span class="st">"pa"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># removing empty labels by setting na.rm = TRUE</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span><span class="op">$</span><span class="va">Family</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 5603</span>
<span class="r-in"><span><span class="va">x3</span> <span class="op">&lt;-</span> <span class="fu">agglomerateByRank</span><span class="op">(</span><span class="va">GlobalPatterns</span>, rank<span class="op">=</span><span class="st">"Family"</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">x3</span><span class="op">)</span> <span class="co"># different from x2</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 341</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Because all the rownames are from the same rank, rownames do not include</span></span></span>
<span class="r-in"><span><span class="co"># prefixes, in this case "Family:".</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">x3</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="op">]</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "125ds10" "211ds20" "5B-12"  </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># To add them, use getTaxonomyLabels function.</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">x3</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="taxonomy-methods.html">getTaxonomyLabels</a></span><span class="op">(</span><span class="va">x3</span>, with.rank <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">x3</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="op">]</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Family:125ds10" "Family:211ds20" "Family:5B-12"  </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># use 'empty.ranks.rm' to remove columns that include only NAs</span></span></span>
<span class="r-in"><span><span class="va">x4</span> <span class="op">&lt;-</span> <span class="fu">agglomerateByRank</span><span class="op">(</span><span class="va">GlobalPatterns</span>, rank<span class="op">=</span><span class="st">"Phylum"</span>,</span></span>
<span class="r-in"><span>                        empty.ranks.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">x4</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataFrame with 6 rows and 2 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     Kingdom          Phylum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 &lt;character&gt;     &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ABY1_OD1           Bacteria        ABY1_OD1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> AC1                Bacteria             AC1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> AD3                Bacteria             AD3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Acidobacteria      Bacteria   Acidobacteria</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Actinobacteria     Bacteria  Actinobacteria</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Armatimonadetes    Bacteria Armatimonadetes</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># If the assay contains NAs, you might want to consider replacing them,</span></span></span>
<span class="r-in"><span><span class="co"># since summing-up NAs lead to NA</span></span></span>
<span class="r-in"><span><span class="va">x5</span> <span class="op">&lt;-</span> <span class="va">GlobalPatterns</span></span></span>
<span class="r-in"><span><span class="co"># Replace first value with NA</span></span></span>
<span class="r-in"><span><span class="fu">assay</span><span class="op">(</span><span class="va">x5</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span></span>
<span class="r-in"><span><span class="va">x6</span> <span class="op">&lt;-</span> <span class="fu">agglomerateByRank</span><span class="op">(</span><span class="va">x5</span>, <span class="st">"Kingdom"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span> <span class="fu">assay</span><span class="op">(</span><span class="va">x6</span><span class="op">)</span> <span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             CL3     CC1    SV1 M31Fcsw M11Fcsw M31Plmr M11Plmr F21Plmr M31Tong</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Archaea      NA    1248  28811      33      57      42     112     140     303</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bacteria 862815 1134209 668698 1543418 2076419  718901  433782  186157 2000099</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          M11Tong LMEpi24M SLEpi20M  AQC1cm  AQC4cm  AQC7cm    NP2     NP3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Archaea       30      131      145    4459   24692   28051   1826   43197</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bacteria  100157  2117461  1217167 1163289 2332489 1671242 521808 1435768</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              NP5 TRRsed1 TRRsed2 TRRsed3   TS28    TS29   Even1  Even2   Even3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Archaea    33996     843    8418   14250   1598    1690     150     23      91</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bacteria 1618758   57845  484708  265454 935868 1209381 1215987 971050 1078150</span>
<span class="r-in"><span><span class="co"># Replace NAs with 0. This is justified when we are summing-up counts.</span></span></span>
<span class="r-in"><span><span class="fu">assay</span><span class="op">(</span><span class="va">x5</span><span class="op">)</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">x5</span><span class="op">)</span><span class="op">)</span> <span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span></span></span>
<span class="r-in"><span><span class="va">x6</span> <span class="op">&lt;-</span> <span class="fu">agglomerateByRank</span><span class="op">(</span><span class="va">x5</span>, <span class="st">"Kingdom"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span> <span class="fu">assay</span><span class="op">(</span><span class="va">x6</span><span class="op">)</span> <span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             CL3     CC1    SV1 M31Fcsw M11Fcsw M31Plmr M11Plmr F21Plmr M31Tong</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Archaea    1262    1248  28811      33      57      42     112     140     303</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bacteria 862815 1134209 668698 1543418 2076419  718901  433782  186157 2000099</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          M11Tong LMEpi24M SLEpi20M  AQC1cm  AQC4cm  AQC7cm    NP2     NP3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Archaea       30      131      145    4459   24692   28051   1826   43197</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bacteria  100157  2117461  1217167 1163289 2332489 1671242 521808 1435768</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              NP5 TRRsed1 TRRsed2 TRRsed3   TS28    TS29   Even1  Even2   Even3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Archaea    33996     843    8418   14250   1598    1690     150     23      91</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bacteria 1618758   57845  484708  265454 935868 1209381 1215987 971050 1078150</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Look at enterotype dataset...</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">enterotype</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">## Print the available taxonomic ranks. Shows only 1 available rank,</span></span></span>
<span class="r-in"><span><span class="co">## not useful for agglomerateByRank</span></span></span>
<span class="r-in"><span><span class="fu"><a href="taxonomy-methods.html">taxonomyRanks</a></span><span class="op">(</span><span class="va">enterotype</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Genus"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### Merge TreeSummarizedExperiments on rows and columns</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">esophagus</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">esophagus</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 58 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(58): 59_8_22 59_5_13 ... 65_9_9 59_2_6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(3): B C D</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: a LinkDataFrame (58 rows)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: 1 phylo tree(s) (58 leaves)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu">rowTree</span><span class="op">(</span><span class="va">esophagus</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="agglomerate-methods-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="co"># get a factor for merging</span></span></span>
<span class="r-in"><span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/regmatches.html" class="external-link">regmatches</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">esophagus</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                       <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">regexpr</a></span><span class="op">(</span><span class="st">"^[0-9]*_[0-9]*"</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">esophagus</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">merged</span> <span class="op">&lt;-</span> <span class="fu">agglomerateByVariable</span><span class="op">(</span><span class="va">esophagus</span>, by <span class="op">=</span> <span class="st">"rows"</span>, <span class="va">f</span>,</span></span>
<span class="r-in"><span>                                update.tree <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu">rowTree</span><span class="op">(</span><span class="va">merged</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="agglomerate-methods-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="co">#</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">GlobalPatterns</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 19216 26 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(19216): 549322 522457 ... 200359 271582</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(26): CL3 CC1 ... Even2 Even3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(7): X.SampleID Primer ... SampleType Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: a LinkDataFrame (19216 rows)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: 1 phylo tree(s) (19216 leaves)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span><span class="va">merged</span> <span class="op">&lt;-</span> <span class="fu">agglomerateByVariable</span><span class="op">(</span><span class="va">GlobalPatterns</span>, by <span class="op">=</span> <span class="st">"cols"</span>,</span></span>
<span class="r-in"><span>                                <span class="fu">colData</span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span><span class="op">$</span><span class="va">SampleType</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">merged</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 19216 9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(19216): 549322 522457 ... 200359 271582</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(9): Feces Freshwater ... Soil Tongue</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(7): X.SampleID Primer ... SampleType Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: a LinkDataFrame (19216 rows)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: 1 phylo tree(s) (19216 leaves)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># print the available taxonomic ranks</span></span></span>
<span class="r-in"><span><span class="fu"><a href="taxonomy-methods.html">taxonomyRanks</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Kingdom" "Phylum"  "Class"   "Order"   "Family"  "Genus"   "Species"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># agglomerateByRanks</span></span></span>
<span class="r-in"><span><span class="co"># </span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">agglomerateByRanks</span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">altExps</span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> List of length 7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> names(7): Kingdom Phylum Class Order Family Genus Species</span>
<span class="r-in"><span><span class="fu">altExp</span><span class="op">(</span><span class="va">tse</span>,<span class="st">"Kingdom"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 2 26 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(1): agglomerated_by_rank</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(2): Archaea Bacteria</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(26): CL3 CC1 ... Even2 Even3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(7): X.SampleID Primer ... SampleType Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: a LinkDataFrame (2 rows)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: 1 phylo tree(s) (19216 leaves)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span><span class="fu">altExp</span><span class="op">(</span><span class="va">tse</span>,<span class="st">"Species"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 944 26 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(1): agglomerated_by_rank</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(944): Abiotrophiadefectiva Achromatiumoxaliferum ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   proteobacteriumsymbiontofOsedaxsp.MB4 symbiontofNoeetapupillata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(26): CL3 CC1 ... Even2 Even3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(7): X.SampleID Primer ... SampleType Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: a LinkDataFrame (944 rows)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: 1 phylo tree(s) (19216 leaves)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># unsplitByRanks</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">unsplitByRanks</span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tse</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 2692 26 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(2692): Kingdom:Archaea Kingdom:Bacteria ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Species:proteobacteriumsymbiontofOsedaxsp.MB4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Species:symbiontofNoeetapupillata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(8): Kingdom Phylum ... Species taxonomicLevel</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(26): CL3 CC1 ... Even2 Even3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(7): X.SampleID Primer ... SampleType Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Felix G.M. Ernst, Sudarshan A. Shetty, Tuomas Borman, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer></div>






  </body></html>

