<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Agglomerate data based on population prevalence — agglomerateByPrevalence • mia</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Agglomerate data based on population prevalence — agglomerateByPrevalence"><meta property="og:description" content="Agglomerate data based on population prevalence"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mia</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.34</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/mia.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/microbiome/mia/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Agglomerate data based on population prevalence</h1>
    <small class="dont-index">Source: <a href="https://github.com/microbiome/mia/blob/devel/R/getPrevalence.R" class="external-link"><code>R/getPrevalence.R</code></a></small>
    <div class="hidden name"><code>agglomerateByPrevalence.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Agglomerate data based on population prevalence</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">agglomerateByPrevalence</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">agglomerateByPrevalence</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  rank <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  other.name <span class="op">=</span> <span class="va">other_label</span>,</span>
<span>  other_label <span class="op">=</span> <span class="st">"Other"</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'TreeSummarizedExperiment'</span></span>
<span><span class="fu">agglomerateByPrevalence</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  rank <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  other.name <span class="op">=</span> <span class="va">other_label</span>,</span>
<span>  other_label <span class="op">=</span> <span class="st">"Other"</span>,</span>
<span>  update.tree <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>a numeric matrix with samples as rows or a
<code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></code>
object.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>arguments passed to <code>agglomerateByRank</code> function for
<code>SummarizedExperiment</code> objects and other functions.
See <code><a href="agglomerate-methods.html">agglomerateByRank</a></code> for more details.</p></dd>


<dt id="arg-rank">rank<a class="anchor" aria-label="anchor" href="#arg-rank"></a></dt>
<dd><p><code>Character scalar</code>. Defines a taxonomic rank. Must be a value of
<code><a href="taxonomy-methods.html">taxonomyRanks()</a></code> function.</p></dd>


<dt id="arg-other-name">other.name<a class="anchor" aria-label="anchor" href="#arg-other-name"></a></dt>
<dd><p><code>Character scalar</code>. Used as the label for the
summary of non-prevalent taxa. (default: <code>"Other"</code>)</p></dd>


<dt id="arg-other-label">other_label<a class="anchor" aria-label="anchor" href="#arg-other-label"></a></dt>
<dd><p>Deprecated. use <code>other.name</code> instead.</p></dd>


<dt id="arg-update-tree">update.tree<a class="anchor" aria-label="anchor" href="#arg-update-tree"></a></dt>
<dd><p><code>Logical scalar</code>. Should
<code>rowTree()</code> also be merged? (Default: <code>FALSE</code>)</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p><code>agglomerateByPrevalence</code> returns a taxonomically-agglomerated object
of the same class as x and based on prevalent taxonomic results.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p><code>agglomerateByPrevalence</code> sums up the values of assays at the taxonomic
level specified by <code>rank</code> (by default the highest taxonomic level
available) and selects the summed results that exceed the given population
prevalence at the given detection level. The other summed values (below the
threshold) are agglomerated in an additional row taking the name indicated by
<code>other.name</code> (by default "Other").</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Data can be aggregated based on prevalent taxonomic results</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="va">GlobalPatterns</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="transformAssay.html">transformAssay</a></span><span class="op">(</span><span class="va">tse</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">agglomerateByPrevalence</span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">tse</span>,</span></span>
<span class="r-in"><span>    rank <span class="op">=</span> <span class="st">"Phylum"</span>,</span></span>
<span class="r-in"><span>    assay.type <span class="op">=</span> <span class="st">"relabundance"</span>,</span></span>
<span class="r-in"><span>    detection <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="fl">100</span>,</span></span>
<span class="r-in"><span>    prevalence <span class="op">=</span> <span class="fl">50</span><span class="op">/</span><span class="fl">100</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">tse</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 6 26 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(2): agglomerated_by_rank agglomerated_by_rank</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(2): counts relabundance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(6): Actinobacteria Bacteroidetes ... Proteobacteria Other</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(26): CL3 CC1 ... Even2 Even3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(7): X.SampleID Primer ... SampleType Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: a LinkDataFrame (6 rows)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: 1 phylo tree(s) (19216 leaves)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Here data is aggregated at the taxonomic level "Phylum". The five phyla</span></span></span>
<span class="r-in"><span><span class="co"># that exceed the population prevalence threshold of 50/100 represent the</span></span></span>
<span class="r-in"><span><span class="co"># five first rows of the assay in the aggregated data. The sixth and last row</span></span></span>
<span class="r-in"><span><span class="co"># named by default "Other" takes the summed up values of all the other phyla</span></span></span>
<span class="r-in"><span><span class="co"># that are below the prevalence threshold.</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">assay</span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   CL3    CC1    SV1 M31Fcsw M11Fcsw</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Actinobacteria  39601  90280 121703    2540     841</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bacteroidetes   67395  96398  93436  804395 1424107</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cyanobacteria    1955   3353  16676     423  212812</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Firmicutes       8584   4726   3524  700084  330423</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Proteobacteria 294228 361327 224004   18798   86614</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Other          452307 579321 238100   17211   21679</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Felix G.M. Ernst, Sudarshan A. Shetty, Tuomas Borman, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer></div>






  </body></html>

