<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Dominant Taxa — dominantTaxa • mia</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Dominant Taxa — dominantTaxa" />
<meta property="og:description" content="These functions return information about the most dominant taxa in a
SummarizedExperiment-class object. Additionally, the information
can be directly stored in the colData." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mia</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.98.34</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/mia.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Dominant Taxa</h1>
    
    <div class="hidden name"><code>dominantTaxa.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>These functions return information about the most dominant taxa in a
<code>SummarizedExperiment-class</code> object. Additionally, the information
can be directly stored in the <code>colData</code>.</p>
    </div>

    <pre class="usage"><span class='fu'>dominantTaxa</span><span class='op'>(</span><span class='va'>x</span>, abund_values <span class='op'>=</span> <span class='st'>"counts"</span>, rank <span class='op'>=</span> <span class='cn'>NULL</span>, name <span class='op'>=</span> <span class='st'>"dominant_taxa"</span><span class='op'>)</span>

<span class='co'># S4 method for SummarizedExperiment</span>
<span class='fu'>dominantTaxa</span><span class='op'>(</span><span class='va'>x</span>, abund_values <span class='op'>=</span> <span class='st'>"counts"</span>, rank <span class='op'>=</span> <span class='cn'>NULL</span>, name <span class='op'>=</span> <span class='st'>"dominant_taxa"</span><span class='op'>)</span>

<span class='fu'>getDominantTaxa</span><span class='op'>(</span>
  <span class='va'>x</span>,
  abund_values <span class='op'>=</span> <span class='st'>"counts"</span>,
  rank <span class='op'>=</span> <span class='cn'>NULL</span>,
  group <span class='op'>=</span> <span class='cn'>NULL</span>,
  name <span class='op'>=</span> <span class='st'>"dominant_taxa"</span>
<span class='op'>)</span>

<span class='co'># S4 method for SummarizedExperiment</span>
<span class='fu'>getDominantTaxa</span><span class='op'>(</span>
  <span class='va'>x</span>,
  abund_values <span class='op'>=</span> <span class='st'>"counts"</span>,
  rank <span class='op'>=</span> <span class='cn'>NULL</span>,
  group <span class='op'>=</span> <span class='cn'>NULL</span>,
  name <span class='op'>=</span> <span class='st'>"dominant_taxa"</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>A <code><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>SummarizedExperiment</a></code>
object.</p></td>
    </tr>
    <tr>
      <th>abund_values</th>
      <td><p>A single character value for selecting the
<code><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></code>
to use for prevalence calculation.</p></td>
    </tr>
    <tr>
      <th>rank</th>
      <td><p>A single character defining a taxonomic rank. Must be a value
of the output of <code><a href='taxonomy-methods.html'>taxonomicRanks()</a></code>.</p></td>
    </tr>
    <tr>
      <th>name</th>
      <td><p>A name for the column of the colData where the dominant taxa
will be stored in. Must be a string.</p></td>
    </tr>
    <tr>
      <th>group</th>
      <td><p>With group, it is possible to group the observations in an overview. Must
be a one of the column names of colData.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>dominantTaxa</code> returns <code>x</code> with additional <code>colData</code> named
<code>*name*</code>. <code>getDominantTaxa</code> returns an overview in a tibble. It contains
dominant taxa in a column named <code>*name*</code> and its abundance in the data set.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p><code>dominantTaxa</code> extracts the most abundant taxa
in a <code>SummarizedExperiment</code> object, and
stores the information in the <code>colData</code>. <code>GetDominantTaxa</code> returns information
about most dominant taxa in a tibble. Information includes their absolute and
relative abundances in whole data set.</p>
<p>With 'rank' parameter, it is possible to agglomerate taxa based on taxonomic ranks.
E.g. if 'family' rank is used, all abundances of same family is added together,
and those families are returned.</p>
<p>With 'group' parameter, it is possible to group observations of returned overview based on
samples' features.  E.g., if samples contain information about patients' health status,
it is possible to group observations, e.g. to 'healthy' and 'sick', and get the most dominant
taxa of different health status.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Leo Lahti and Tuomas Borman. Contact: <a href='microbiome.github.io'>microbiome.github.io</a></p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>GlobalPatterns</span><span class='op'>)</span>
<span class='va'>x</span> <span class='op'>&lt;-</span> <span class='va'>GlobalPatterns</span>

<span class='co'># Finds the dominant taxa.</span>
<span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'>dominantTaxa</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
<span class='co'># Information is stored to colData</span>
<span class='fu'>colData</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
</div><div class='output co'>#&gt; DataFrame with 26 rows and 8 columns
#&gt;         X.SampleID   Primer Final_Barcode Barcode_truncated_plus_T
#&gt;           &lt;factor&gt; &lt;factor&gt;      &lt;factor&gt;                 &lt;factor&gt;
#&gt; CL3        CL3      ILBC_01        AACGCA                   TGCGTT
#&gt; CC1        CC1      ILBC_02        AACTCG                   CGAGTT
#&gt; SV1        SV1      ILBC_03        AACTGT                   ACAGTT
#&gt; M31Fcsw    M31Fcsw  ILBC_04        AAGAGA                   TCTCTT
#&gt; M11Fcsw    M11Fcsw  ILBC_05        AAGCTG                   CAGCTT
#&gt; ...            ...      ...           ...                      ...
#&gt; TS28         TS28   ILBC_25        ACCAGA                   TCTGGT
#&gt; TS29         TS29   ILBC_26        ACCAGC                   GCTGGT
#&gt; Even1        Even1  ILBC_27        ACCGCA                   TGCGGT
#&gt; Even2        Even2  ILBC_28        ACCTCG                   CGAGGT
#&gt; Even3        Even3  ILBC_29        ACCTGT                   ACAGGT
#&gt;         Barcode_full_length SampleType
#&gt;                    &lt;factor&gt;   &lt;factor&gt;
#&gt; CL3             CTAGCGTGCGT      Soil 
#&gt; CC1             CATCGACGAGT      Soil 
#&gt; SV1             GTACGCACAGT      Soil 
#&gt; M31Fcsw         TCGACATCTCT      Feces
#&gt; M11Fcsw         CGACTGCAGCT      Feces
#&gt; ...                     ...        ...
#&gt; TS28            GCATCGTCTGG      Feces
#&gt; TS29            CTAGTCGCTGG      Feces
#&gt; Even1           TGACTCTGCGG      Mock 
#&gt; Even2           TCTGATCGAGG      Mock 
#&gt; Even3           AGAGAGACAGG      Mock 
#&gt;                                        Description dominant_taxa
#&gt;                                           &lt;factor&gt;   &lt;character&gt;
#&gt; CL3     Calhoun South Carolina Pine soil, pH 4.9           36155
#&gt; CC1     Cedar Creek Minnesota, grassland, pH 6.1          256977
#&gt; SV1     Sevilleta new Mexico, desert scrub, pH 8.3         71074
#&gt; M31Fcsw M3, Day 1, fecal swab, whole body study           331820
#&gt; M11Fcsw M1, Day 1, fecal swab, whole body study           331820
#&gt; ...                                            ...           ...
#&gt; TS28                                       Twin #1        331820
#&gt; TS29                                       Twin #2        189047
#&gt; Even1                                      Even1          550960
#&gt; Even2                                      Even2          550960
#&gt; Even3                                      Even3          550960</div><div class='input'><span class='co'>#Gets the overview of dominant taxa</span>
<span class='va'>overview</span> <span class='op'>&lt;-</span> <span class='fu'>getDominantTaxa</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
<span class='va'>overview</span>
</div><div class='output co'>#&gt; <span style='color: #555555;'># A tibble: 19 x 4</span><span>
#&gt;    dominant_taxa     n rel.freq rel.freq.pct
#&gt;    </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>         </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>    </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>       
#&gt; </span><span style='color: #555555;'> 1</span><span> 331820            3     11.5 12%         
#&gt; </span><span style='color: #555555;'> 2</span><span> 549656            3     11.5 12%         
#&gt; </span><span style='color: #555555;'> 3</span><span> 550960            3     11.5 12%         
#&gt; </span><span style='color: #555555;'> 4</span><span> 319044            2      7.7 8%          
#&gt; </span><span style='color: #555555;'> 5</span><span> 114821            1      3.8 4%          
#&gt; </span><span style='color: #555555;'> 6</span><span> 12812             1      3.8 4%          
#&gt; </span><span style='color: #555555;'> 7</span><span> 189047            1      3.8 4%          
#&gt; </span><span style='color: #555555;'> 8</span><span> 256977            1      3.8 4%          
#&gt; </span><span style='color: #555555;'> 9</span><span> 279599            1      3.8 4%          
#&gt; </span><span style='color: #555555;'>10</span><span> 329744            1      3.8 4%          
#&gt; </span><span style='color: #555555;'>11</span><span> 360229            1      3.8 4%          
#&gt; </span><span style='color: #555555;'>12</span><span> 36155             1      3.8 4%          
#&gt; </span><span style='color: #555555;'>13</span><span> 484436            1      3.8 4%          
#&gt; </span><span style='color: #555555;'>14</span><span> 534609            1      3.8 4%          
#&gt; </span><span style='color: #555555;'>15</span><span> 557211            1      3.8 4%          
#&gt; </span><span style='color: #555555;'>16</span><span> 64396             1      3.8 4%          
#&gt; </span><span style='color: #555555;'>17</span><span> 71074             1      3.8 4%          
#&gt; </span><span style='color: #555555;'>18</span><span> 87194             1      3.8 4%          
#&gt; </span><span style='color: #555555;'>19</span><span> 98605             1      3.8 4%          </div><div class='input'>
<span class='co'># If taxonomic information is available, it is possible to find the most dominant</span>
<span class='co'>#group from specific taxonomic level, here family level. The name of column can be specified.</span>
<span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'>dominantTaxa</span><span class='op'>(</span><span class='va'>x</span>, rank<span class='op'>=</span><span class='st'>"Family"</span>, name<span class='op'>=</span><span class='st'>"dominant_taxa_ranked_with_family"</span><span class='op'>)</span>
<span class='fu'>colData</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
</div><div class='output co'>#&gt; DataFrame with 26 rows and 9 columns
#&gt;         X.SampleID   Primer Final_Barcode Barcode_truncated_plus_T
#&gt;           &lt;factor&gt; &lt;factor&gt;      &lt;factor&gt;                 &lt;factor&gt;
#&gt; CL3        CL3      ILBC_01        AACGCA                   TGCGTT
#&gt; CC1        CC1      ILBC_02        AACTCG                   CGAGTT
#&gt; SV1        SV1      ILBC_03        AACTGT                   ACAGTT
#&gt; M31Fcsw    M31Fcsw  ILBC_04        AAGAGA                   TCTCTT
#&gt; M11Fcsw    M11Fcsw  ILBC_05        AAGCTG                   CAGCTT
#&gt; ...            ...      ...           ...                      ...
#&gt; TS28         TS28   ILBC_25        ACCAGA                   TCTGGT
#&gt; TS29         TS29   ILBC_26        ACCAGC                   GCTGGT
#&gt; Even1        Even1  ILBC_27        ACCGCA                   TGCGGT
#&gt; Even2        Even2  ILBC_28        ACCTCG                   CGAGGT
#&gt; Even3        Even3  ILBC_29        ACCTGT                   ACAGGT
#&gt;         Barcode_full_length SampleType
#&gt;                    &lt;factor&gt;   &lt;factor&gt;
#&gt; CL3             CTAGCGTGCGT      Soil 
#&gt; CC1             CATCGACGAGT      Soil 
#&gt; SV1             GTACGCACAGT      Soil 
#&gt; M31Fcsw         TCGACATCTCT      Feces
#&gt; M11Fcsw         CGACTGCAGCT      Feces
#&gt; ...                     ...        ...
#&gt; TS28            GCATCGTCTGG      Feces
#&gt; TS29            CTAGTCGCTGG      Feces
#&gt; Even1           TGACTCTGCGG      Mock 
#&gt; Even2           TCTGATCGAGG      Mock 
#&gt; Even3           AGAGAGACAGG      Mock 
#&gt;                                        Description dominant_taxa
#&gt;                                           &lt;factor&gt;   &lt;character&gt;
#&gt; CL3     Calhoun South Carolina Pine soil, pH 4.9           36155
#&gt; CC1     Cedar Creek Minnesota, grassland, pH 6.1          256977
#&gt; SV1     Sevilleta new Mexico, desert scrub, pH 8.3         71074
#&gt; M31Fcsw M3, Day 1, fecal swab, whole body study           331820
#&gt; M11Fcsw M1, Day 1, fecal swab, whole body study           331820
#&gt; ...                                            ...           ...
#&gt; TS28                                       Twin #1        331820
#&gt; TS29                                       Twin #2        189047
#&gt; Even1                                      Even1          550960
#&gt; Even2                                      Even2          550960
#&gt; Even3                                      Even3          550960
#&gt;         dominant_taxa_ranked_with_family
#&gt;                              &lt;character&gt;
#&gt; CL3                      Solibacteraceae
#&gt; CC1                   Spartobacteriaceae
#&gt; SV1                    Sphingomonadaceae
#&gt; M31Fcsw                   Bacteroidaceae
#&gt; M11Fcsw                   Bacteroidaceae
#&gt; ...                                  ...
#&gt; TS28                     Ruminococcaceae
#&gt; TS29                     Ruminococcaceae
#&gt; Even1                    Lachnospiraceae
#&gt; Even2                 Enterobacteriaceae
#&gt; Even3                    Lachnospiraceae</div><div class='input'><span class='co'>#Gets the overview of dominant taxa</span>
<span class='va'>overview</span> <span class='op'>&lt;-</span> <span class='fu'>getDominantTaxa</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
<span class='va'>overview</span>
</div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 19 x 4</span><span>
#&gt;    dominant_taxa     n rel.freq rel.freq.pct
#&gt;    </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>         </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>    </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>       
#&gt; </span><span style='color: #555555;'> 1</span><span> 331820            3     11.5 12%         
#&gt; </span><span style='color: #555555;'> 2</span><span> 549656            3     11.5 12%         
#&gt; </span><span style='color: #555555;'> 3</span><span> 550960            3     11.5 12%         
#&gt; </span><span style='color: #555555;'> 4</span><span> 319044            2      7.7 8%          
#&gt; </span><span style='color: #555555;'> 5</span><span> 114821            1      3.8 4%          
#&gt; </span><span style='color: #555555;'> 6</span><span> 12812             1      3.8 4%          
#&gt; </span><span style='color: #555555;'> 7</span><span> 189047            1      3.8 4%          
#&gt; </span><span style='color: #555555;'> 8</span><span> 256977            1      3.8 4%          
#&gt; </span><span style='color: #555555;'> 9</span><span> 279599            1      3.8 4%          
#&gt; </span><span style='color: #555555;'>10</span><span> 329744            1      3.8 4%          
#&gt; </span><span style='color: #555555;'>11</span><span> 360229            1      3.8 4%          
#&gt; </span><span style='color: #555555;'>12</span><span> 36155             1      3.8 4%          
#&gt; </span><span style='color: #555555;'>13</span><span> 484436            1      3.8 4%          
#&gt; </span><span style='color: #555555;'>14</span><span> 534609            1      3.8 4%          
#&gt; </span><span style='color: #555555;'>15</span><span> 557211            1      3.8 4%          
#&gt; </span><span style='color: #555555;'>16</span><span> 64396             1      3.8 4%          
#&gt; </span><span style='color: #555555;'>17</span><span> 71074             1      3.8 4%          
#&gt; </span><span style='color: #555555;'>18</span><span> 87194             1      3.8 4%          
#&gt; </span><span style='color: #555555;'>19</span><span> 98605             1      3.8 4%          </div><div class='input'>
<span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'>microbiomeDataSets</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/microbiomeDataSets/man/OKeefeDSData.html'>dietswap</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>using temporary cache /tmp/Rtmpr6XY74/BiocFileCache</span></div><div class='output co'>#&gt; <span class='message'>snapshotDate(): 2021-02-09</span></div><div class='output co'>#&gt; <span class='message'>see ?microbiomeDataSets and browseVignettes('microbiomeDataSets') for documentation</span></div><div class='output co'>#&gt; <span class='message'>downloading 1 resources</span></div><div class='output co'>#&gt; <span class='message'>retrieving 1 resource</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='output co'>#&gt; <span class='message'>see ?microbiomeDataSets and browseVignettes('microbiomeDataSets') for documentation</span></div><div class='output co'>#&gt; <span class='message'>downloading 1 resources</span></div><div class='output co'>#&gt; <span class='message'>retrieving 1 resource</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='output co'>#&gt; <span class='message'>see ?microbiomeDataSets and browseVignettes('microbiomeDataSets') for documentation</span></div><div class='output co'>#&gt; <span class='message'>downloading 1 resources</span></div><div class='output co'>#&gt; <span class='message'>retrieving 1 resource</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'><span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'>dominantTaxa</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
<span class='fu'>colData</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
</div><div class='output co'>#&gt; DataFrame with 222 rows and 9 columns
#&gt;             subject      sex nationality    group      sample timepoint
#&gt;            &lt;factor&gt; &lt;factor&gt;    &lt;factor&gt; &lt;factor&gt; &lt;character&gt; &lt;integer&gt;
#&gt; Sample-1        byn   male           AAM       DI    Sample-1         4
#&gt; Sample-2        nms   male           AFR       HE    Sample-2         2
#&gt; Sample-3        olt   male           AFR       HE    Sample-3         2
#&gt; Sample-4        pku   female         AFR       HE    Sample-4         2
#&gt; Sample-5        qjy   female         AFR       HE    Sample-5         2
#&gt; ...             ...      ...         ...      ...         ...       ...
#&gt; Sample-218      qjy   female         AFR       ED  Sample-218         6
#&gt; Sample-219      riv   female         AFR       ED  Sample-219         6
#&gt; Sample-220      shj   female         AFR       ED  Sample-220         6
#&gt; Sample-221      tgx   male           AFR       ED  Sample-221         6
#&gt; Sample-222      ufm   male           AFR       ED  Sample-222         6
#&gt;            timepoint.within.group  bmi_group          dominant_taxa
#&gt;                         &lt;integer&gt;   &lt;factor&gt;            &lt;character&gt;
#&gt; Sample-1                        1 obese      Bacteroides vulgatus..
#&gt; Sample-2                        1 lean       Prevotella melaninog..
#&gt; Sample-3                        1 overweight Prevotella melaninog..
#&gt; Sample-4                        1 obese      Oscillospira guiller..
#&gt; Sample-5                        1 overweight Prevotella melaninog..
#&gt; ...                           ...        ...                    ...
#&gt; Sample-218                      2 overweight Oscillospira guiller..
#&gt; Sample-219                      2 obese      Prevotella melaninog..
#&gt; Sample-220                      2 obese      Prevotella melaninog..
#&gt; Sample-221                      2 overweight Oscillospira guiller..
#&gt; Sample-222                      2 lean       Prevotella melaninog..</div><div class='input'><span class='co'># With group, it is possible to group observations based on group specified by user</span>
<span class='co'>#Gets the overview of dominant taxa</span>
<span class='va'>overview</span> <span class='op'>&lt;-</span> <span class='fu'>getDominantTaxa</span><span class='op'>(</span><span class='va'>x</span>, group <span class='op'>=</span> <span class='st'>"nationality"</span><span class='op'>)</span>
<span class='va'>overview</span>
</div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 14 x 5</span><span>
#&gt; </span><span style='color: #555555;'># Groups:   nationality [2]</span><span>
#&gt;    nationality dominant_taxa                            n rel.freq rel.freq.pct
#&gt;    </span><span style='color: #555555;font-style: italic;'>&lt;fct&gt;</span><span>       </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>                                </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>    </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>       
#&gt; </span><span style='color: #555555;'> 1</span><span> AFR         Prevotella melaninogenica et rel.       69     69.7 70%         
#&gt; </span><span style='color: #555555;'> 2</span><span> AAM         Bacteroides vulgatus et rel.            57     46.3 46%         
#&gt; </span><span style='color: #555555;'> 3</span><span> AAM         Prevotella melaninogenica et rel.       35     28.5 28%         
#&gt; </span><span style='color: #555555;'> 4</span><span> AFR         Oscillospira guillermondii et rel.      26     26.3 26%         
#&gt; </span><span style='color: #555555;'> 5</span><span> AAM         Oscillospira guillermondii et rel.      17     13.8 14%         
#&gt; </span><span style='color: #555555;'> 6</span><span> AAM         Clostridium cellulosi et rel.            7      5.7 6%          
#&gt; </span><span style='color: #555555;'> 7</span><span> AAM         Akkermansia                              2      1.6 2%          
#&gt; </span><span style='color: #555555;'> 8</span><span> AAM         Dialister                                2      1.6 2%          
#&gt; </span><span style='color: #555555;'> 9</span><span> AFR         Clostridium cellulosi et rel.            2      2   2%          
#&gt; </span><span style='color: #555555;'>10</span><span> AAM         Faecalibacterium prausnitzii et rel.     1      0.8 1%          
#&gt; </span><span style='color: #555555;'>11</span><span> AAM         Ruminococcus obeum et rel.               1      0.8 1%          
#&gt; </span><span style='color: #555555;'>12</span><span> AAM         Uncultured Clostridiales I               1      0.8 1%          
#&gt; </span><span style='color: #555555;'>13</span><span> AFR         Faecalibacterium prausnitzii et rel.     1      1   1%          
#&gt; </span><span style='color: #555555;'>14</span><span> AFR         Streptococcus bovis et rel.              1      1   1%          </div><div class='input'>
</div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Felix G.M. Ernst, Sudarshan Shetty, Leo Lahti, Tuomas Borman.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


