<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Import QIIME2 results to TreeSummarizedExperiment — importQIIME2 • mia</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Import QIIME2 results to TreeSummarizedExperiment — importQIIME2"><meta property="og:description" content="Results exported from QIMME2 can be imported as a
TreeSummarizedExperiment using importQIIME2. Except for the
assay.file, the other data types, row.file,
refseq.file and tree.file, are optional, but are highly
encouraged to be provided.
Import the QIIME2 artifacts to R."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mia</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.31</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/mia.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/microbiome/mia/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Import QIIME2 results to <code>TreeSummarizedExperiment</code></h1>
    <small class="dont-index">Source: <a href="https://github.com/microbiome/mia/blob/devel/R/importQIIME2.R" class="external-link"><code>R/importQIIME2.R</code></a></small>
    <div class="hidden name"><code>importQIIME2.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Results exported from QIMME2 can be imported as a
<code>TreeSummarizedExperiment</code> using <code>importQIIME2</code>. Except for the
<code>assay.file</code>, the other data types, <code>row.file</code>,
<code>refseq.file</code> and <code>tree.file</code>, are optional, but are highly
encouraged to be provided.</p>
<p>Import the QIIME2 artifacts to R.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">importQIIME2</span><span class="op">(</span></span>
<span>  assay.file <span class="op">=</span> <span class="va">featureTableFile</span>,</span>
<span>  <span class="va">featureTableFile</span>,</span>
<span>  row.file <span class="op">=</span> <span class="va">taxonomyTableFile</span>,</span>
<span>  taxonomyTableFile <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  col.file <span class="op">=</span> <span class="va">sampleMetaFile</span>,</span>
<span>  sampleMetaFile <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  as.refseq <span class="op">=</span> <span class="va">featureNamesAsRefSeq</span>,</span>
<span>  featureNamesAsRefSeq <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  refseq.file <span class="op">=</span> <span class="va">refSeqFile</span>,</span>
<span>  refSeqFile <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tree.file <span class="op">=</span> <span class="va">phyTreeFile</span>,</span>
<span>  phyTreeFile <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">importQZA</span><span class="op">(</span><span class="va">file</span>, temp.dir <span class="op">=</span> <span class="va">temp</span>, temp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-assay-file">assay.file<a class="anchor" aria-label="anchor" href="#arg-assay-file"></a></dt>
<dd><p>a single <code>character</code> value defining the file
path of the feature table to be imported.</p></dd>


<dt id="arg-featuretablefile">featureTableFile<a class="anchor" aria-label="anchor" href="#arg-featuretablefile"></a></dt>
<dd><p>Deprecated. use <code>assay.file</code> instead.</p></dd>


<dt id="arg-row-file">row.file<a class="anchor" aria-label="anchor" href="#arg-row-file"></a></dt>
<dd><p>a single <code>character</code> value defining the file
path of the taxonomy table to be imported. (default:
<code>row.file = NULL</code>).</p></dd>


<dt id="arg-taxonomytablefile">taxonomyTableFile<a class="anchor" aria-label="anchor" href="#arg-taxonomytablefile"></a></dt>
<dd><p>Deprecated. use <code>row.file</code> instead.</p></dd>


<dt id="arg-col-file">col.file<a class="anchor" aria-label="anchor" href="#arg-col-file"></a></dt>
<dd><p>a single <code>character</code> value defining the file path
of the sample metadata to be imported. The file has to be in tsv format.
(default: <code>col.file = NULL</code>).</p></dd>


<dt id="arg-samplemetafile">sampleMetaFile<a class="anchor" aria-label="anchor" href="#arg-samplemetafile"></a></dt>
<dd><p>Deprecated. Use <code>col.file</code> instead.</p></dd>


<dt id="arg-as-refseq">as.refseq<a class="anchor" aria-label="anchor" href="#arg-as-refseq"></a></dt>
<dd><p><code>TRUE</code> or <code>FALSE</code>: Should the feature
names of the feature table be regarded as reference sequences? This setting
will be disregarded, if <code>refseq.file</code> is not <code>NULL</code>. If the
feature names do not contain valid DNA characters only, the reference
sequences will not be set.</p></dd>


<dt id="arg-featurenamesasrefseq">featureNamesAsRefSeq<a class="anchor" aria-label="anchor" href="#arg-featurenamesasrefseq"></a></dt>
<dd><p>Deprecated. Use <code>as.refseq</code> instead.</p></dd>


<dt id="arg-refseq-file">refseq.file<a class="anchor" aria-label="anchor" href="#arg-refseq-file"></a></dt>
<dd><p>a single <code>character</code> value defining the file path of
the reference sequences for each feature. (default: <code>refseq.file =
  NULL</code>).</p></dd>


<dt id="arg-refseqfile">refSeqFile<a class="anchor" aria-label="anchor" href="#arg-refseqfile"></a></dt>
<dd><p>Deprecated. Use <code>refseq.file</code> instead.</p></dd>


<dt id="arg-tree-file">tree.file<a class="anchor" aria-label="anchor" href="#arg-tree-file"></a></dt>
<dd><p>a single <code>character</code> value defining the file path of
the phylogenetic tree. (default: <code>tree.file = NULL</code>).</p></dd>


<dt id="arg-phytreefile">phyTreeFile<a class="anchor" aria-label="anchor" href="#arg-phytreefile"></a></dt>
<dd><p>Deprecated. Use <code>tree.file</code> isntead.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>additional arguments:</p><ul><li><p><code>temp.dir</code>: the temporary directory used for decompressing the
data. (default: <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>)</p></li>
<li><p><code>prefix.rm</code>: <code>TRUE</code> or <code>FALSE</code>: Should
taxonomic prefixes be removed? (default:
<code>prefix.rm = FALSE</code>)</p></li>
</ul></dd>


<dt id="arg-file">file<a class="anchor" aria-label="anchor" href="#arg-file"></a></dt>
<dd><p>character, path of the input qza file. Only files in format of
<code>BIOMV210DirFmt</code> (feature table), <code>TSVTaxonomyDirectoryFormat</code> (taxonomic
table), <code>NewickDirectoryFormat</code> (phylogenetic tree ) and
<code>DNASequencesDirectoryFormat</code> (representative sequences) are supported
right now.</p></dd>


<dt id="arg-temp-dir">temp.dir<a class="anchor" aria-label="anchor" href="#arg-temp-dir"></a></dt>
<dd><p>character, a temporary directory in which the qza file will be
decompressed to, default <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>.</p></dd>


<dt id="arg-temp">temp<a class="anchor" aria-label="anchor" href="#arg-temp"></a></dt>
<dd><p>Deprecated. Use <code>temp.dir</code> isntead.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A
<code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code>
object</p>
<p><code>matrix</code> object for feature table, <code>DataFrame</code> for taxonomic table,
<code><a href="https://rdrr.io/pkg/ape/man/read.tree.html" class="external-link">ape::phylo</a></code> object for phylogenetic tree,
<code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">Biostrings::DNAStringSet</a></code> for representative sequences of taxa.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Both arguments <code>as.refseq</code> and <code>refseq.file</code> can be used
to define reference sequences of features. <code>as.refseq</code> is
only taken into account, if <code>refseq.file</code> is <code>NULL</code>. No reference
sequences are tried to be created, if <code>featureNameAsRefSeq</code> is
<code>FALSE</code> and <code>refseq.file</code> is <code>NULL</code>.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Bolyen E et al. 2019: Reproducible, interactive, scalable and extensible
microbiome data science using QIIME 2. Nature Biotechnology 37: 852–857.
<a href="https://doi.org/10.1038/s41587-019-0209-9" class="external-link">https://doi.org/10.1038/s41587-019-0209-9</a></p>
<p><a href="https://qiime2.org" class="external-link">https://qiime2.org</a></p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="convert.html">convertFromPhyloseq</a></code>
<code><a href="convert.html">convertFromBIOM</a></code>
<code><a href="convert.html">convertFromDADA2</a></code>
<code><a href="importMothur.html">importMothur</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Yang Cao</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">assay.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"table.qza"</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">row.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"taxonomy.qza"</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">col.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sample-metadata.tsv"</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tree.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"tree.qza"</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">refseq.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"refseq.qza"</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">importQIIME2</span><span class="op">(</span></span></span>
<span class="r-in"><span>  assay.file <span class="op">=</span> <span class="va">assay.file</span>,</span></span>
<span class="r-in"><span>  row.file <span class="op">=</span> <span class="va">row.file</span>,</span></span>
<span class="r-in"><span>  col.file <span class="op">=</span> <span class="va">col.file</span>,</span></span>
<span class="r-in"><span>  refseq.file <span class="op">=</span> <span class="va">refseq.file</span>,</span></span>
<span class="r-in"><span>  tree.file <span class="op">=</span> <span class="va">tree.file</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">tse</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 770 34 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(770): 4b5eeb300368260019c1fbc7a3c718fc</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   fe30ff0f71a38a39cf1717ec2be3a2fc ... 98d250a339a635f20e26397dafc6ced3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   1830c14ead81ad012f1db0e12f8ab6a4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(8): kingdom phylum ... species Confidence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(34): L1S8 L1S57 ... L6S68 L6S93</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(9): sample.id barcode.sequence ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   reported.antibiotic.usage days.since.experiment.start</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: a LinkDataFrame (770 rows)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: 1 phylo tree(s) (770 leaves)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> referenceSeq: a DNAStringSet (770 sequences)</span>
<span class="r-in"><span><span class="co"># Read individual files</span></span></span>
<span class="r-in"><span><span class="va">assay.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"table.qza"</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">row.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"taxonomy.qza"</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">col.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sample-metadata.tsv"</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">assay</span> <span class="op">&lt;-</span> <span class="fu">importQZA</span><span class="op">(</span><span class="va">assay.file</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">rowdata</span> <span class="op">&lt;-</span> <span class="fu">importQZA</span><span class="op">(</span><span class="va">row.file</span>, prefix.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">coldata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">col.file</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, comment.char <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Assign rownames </span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">coldata</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">coldata</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Order coldata based on assay</span></span></span>
<span class="r-in"><span><span class="va">coldata</span> <span class="op">&lt;-</span> <span class="va">coldata</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">assay</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Create SE from individual files</span></span></span>
<span class="r-in"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></span><span class="op">(</span>assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">assay</span><span class="op">)</span>, rowData <span class="op">=</span> <span class="va">rowdata</span>, colData <span class="op">=</span> <span class="va">coldata</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">se</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: SummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 770 34 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(770): 4b5eeb300368260019c1fbc7a3c718fc</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   fe30ff0f71a38a39cf1717ec2be3a2fc ... 98d250a339a635f20e26397dafc6ced3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   1830c14ead81ad012f1db0e12f8ab6a4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(8): kingdom phylum ... species Confidence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(34): L1S105 L1S140 ... L6S68 L6S93</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(8): barcode.sequence body.site ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   reported.antibiotic.usage days.since.experiment.start</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Felix G.M. Ernst, Sudarshan A. Shetty, Tuomas Borman, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer></div>






  </body></html>

