<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Merge SE objects into single SE object. — mergeSEs • mia</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Merge SE objects into single SE object. — mergeSEs"><meta property="og:description" content="Merge SE objects into single SE object."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mia</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.32</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/mia.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/microbiome/mia/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Merge SE objects into single SE object.</h1>
    <small class="dont-index">Source: <a href="https://github.com/microbiome/mia/blob/devel/R/mergeSEs.R" class="external-link"><code>R/mergeSEs.R</code></a></small>
    <div class="hidden name"><code>mergeSEs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Merge SE objects into single SE object.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">mergeSEs</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SimpleList'</span></span>
<span><span class="fu">mergeSEs</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  assay_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  join <span class="op">=</span> <span class="st">"full"</span>,</span>
<span>  missing.values <span class="op">=</span> <span class="va">missing_values</span>,</span>
<span>  missing_values <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  collapse.cols <span class="op">=</span> <span class="va">collapse_samples</span>,</span>
<span>  collapse_samples <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  collapse.rows <span class="op">=</span> <span class="va">collapse_features</span>,</span>
<span>  collapse_features <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">mergeSEs</span><span class="op">(</span><span class="va">x</span>, y <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'list'</span></span>
<span><span class="fu">mergeSEs</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>a <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></code> object or a list of
<code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></code> objects.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>optional arguments (not used).</p></dd>


<dt id="arg-assay-type">assay.type<a class="anchor" aria-label="anchor" href="#arg-assay-type"></a></dt>
<dd><p>A character value for selecting the
<code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></code>
to be merged. (By default: <code>assay.type = "counts"</code>)</p></dd>


<dt id="arg-assay-name">assay_name<a class="anchor" aria-label="anchor" href="#arg-assay-name"></a></dt>
<dd><p>(Deprecated) alias for <code>assay.type</code>.</p></dd>


<dt id="arg-join">join<a class="anchor" aria-label="anchor" href="#arg-join"></a></dt>
<dd><p>A single character value for selecting the joining method.
Must be 'full', 'inner', 'left', or 'right'. 'left' and 'right' are disabled
when more than two objects are being merged.  (By default: <code>join = "full"</code>)</p></dd>


<dt id="arg-missing-values">missing.values<a class="anchor" aria-label="anchor" href="#arg-missing-values"></a></dt>
<dd><p>NA, 0, or a single character values specifying the notation
of missing values. (By default: <code>missing.values = NA</code>)</p></dd>


<dt id="arg-missing-values">missing_values<a class="anchor" aria-label="anchor" href="#arg-missing-values"></a></dt>
<dd><p>Deprecated. Use <code>missing.values</code> instead.</p></dd>


<dt id="arg-collapse-cols">collapse.cols<a class="anchor" aria-label="anchor" href="#arg-collapse-cols"></a></dt>
<dd><p>A boolean value for selecting whether to collapse identically
named samples to one. (By default: <code>collapse.cols = FALSE</code>)</p></dd>


<dt id="arg-collapse-samples">collapse_samples<a class="anchor" aria-label="anchor" href="#arg-collapse-samples"></a></dt>
<dd><p>Deprecated. Use <code>collapse.cols</code> instead.</p></dd>


<dt id="arg-collapse-rows">collapse.rows<a class="anchor" aria-label="anchor" href="#arg-collapse-rows"></a></dt>
<dd><p>A boolean value for selecting whether to collapse identically
named features to one. Since all taxonomy information is taken into account,
this concerns rownames-level (usually strain level) comparison. Often
OTU or ASV level is just an arbitrary number series from sequencing machine
meaning that the OTU information is not comparable between studies. With this
option, it is possible to specify whether these strains are combined if their
taxonomy information along with OTU number matches.
(By default: <code>collapse.rows = TRUE</code>)</p></dd>


<dt id="arg-collapse-features">collapse_features<a class="anchor" aria-label="anchor" href="#arg-collapse-features"></a></dt>
<dd><p>Deprecated. Use <code>collapse.rows</code> instead.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>A single boolean value to choose whether to show messages.
(By default: <code>verbose = TRUE</code>)</p></dd>


<dt id="arg-y">y<a class="anchor" aria-label="anchor" href="#arg-y"></a></dt>
<dd><p>a <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></code> object when <code>x</code> is a
<code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></code> object. Disabled when <code>x</code> is a list.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A single <code>SummarizedExperiment</code> object.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This function merges multiple <code>SummarizedExperiment</code> objects. It combines
<code>rowData</code>, <code>assays</code>, and <code>colData</code> so that the output includes
each unique row and column ones. The merging is done based on <code>rownames</code> and
<code>colnames</code>. <code>rowTree</code> and <code>colTree</code> are preserved if linkage
between rows/cols and the tree is found.</p>
<p>Equally named rows are interpreted as equal. Further
matching based on <code>rowData</code> is not done. For samples, collapsing
is disabled by default meaning that equally named samples that are stored
in different objects are interpreted as unique. Collapsing can be enabled
with <code>collapse.cols = TRUE</code> when equally named samples describe the same
sample.</p>
<p>If, for example, all rows are not shared with
individual objects, there are missing values in <code>assays</code>. The notation of missing
can be specified with the <code>missing.values</code> argument. If input consists of
<code>TreeSummarizedExperiment</code> objects, also <code>rowTree</code>, <code>colTree</code>, and
<code>referenceSeq</code> are preserved if possible. The data is preserved if
all the rows or columns can be found from it.</p>
<p>Compared to <code>cbind</code> and <code>rbind</code> <code>mergeSEs</code>
allows more freely merging since <code>cbind</code> and <code>rbind</code> expect
that rows and columns are matching, respectively.</p>
<p>You can choose joining methods from <code>'full'</code>, <code>'inner'</code>,
<code>'left'</code>, and  <code>'right'</code>. In all the methods, all the samples are
included in the result object. However, with different methods, it is possible
to choose which rows are included.</p>
<ul><li><p><code>full</code> – all unique features</p></li>
<li><p><code>inner</code> – all shared features</p></li>
<li><p><code>left</code> – all the features of the first object</p></li>
<li><p><code>right</code> – all the features of the second object</p></li>
</ul><p>The output depends on the input. If the input contains <code>SummarizedExperiment</code>
object, then the output will be <code>SummarizedExperiment</code>. When all the input
objects belong to <code>TreeSummarizedExperiment</code>, the output will be
<code>TreeSummarizedExperiment</code>.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index">
<ul><li><p><code>TreeSummarizedExperiment::cbind</code></p></li>
<li><p><code>TreeSummarizedExperiment::rbind</code></p></li>
<li><p><code><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">full_join</a></code></p></li>
<li><p><code><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">inner_join</a></code></p></li>
<li><p><code><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></code></p></li>
<li><p><code><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">right_join</a></code></p></li>
</ul></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Leo Lahti and Tuomas Borman. Contact: <a href="microbiome.github.io">microbiome.github.io</a></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">esophagus</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">enterotype</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Take only subsets so that it wont take so long</span></span></span>
<span class="r-in"><span><span class="va">tse1</span> <span class="op">&lt;-</span> <span class="va">GlobalPatterns</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span>, <span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">tse2</span> <span class="op">&lt;-</span> <span class="va">esophagus</span></span></span>
<span class="r-in"><span><span class="va">tse3</span> <span class="op">&lt;-</span> <span class="va">enterotype</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span>, <span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Merge two TreeSEs</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">mergeSEs</span><span class="op">(</span><span class="va">tse1</span>, <span class="va">tse2</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Merging with full join...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1/2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2/2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Merging rowTree...</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Merge a list of TreeSEs</span></span></span>
<span class="r-in"><span><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu">SimpleList</span><span class="op">(</span><span class="va">tse1</span>, <span class="va">tse2</span>, <span class="va">tse3</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">mergeSEs</span><span class="op">(</span><span class="va">list</span>, assay.type <span class="op">=</span> <span class="st">"counts"</span>, missing.values <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Merging with full join...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1/3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2/3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3/3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Merging rowTree...</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>rowTree(s) does not match with the data so it is discarded.</span>
<span class="r-in"><span><span class="va">tse</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 258 309 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(258): Abiotrophia Achromobacter ... 9_7_25 Bacteria</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(309): AM.AD.1 AM.AD.2 ... TS99.2_V2 TS9_V2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(16): X.SampleID Primer ... Age ClinicalStatus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># With 'join', it is possible to specify the merging method. Subsets are used</span></span></span>
<span class="r-in"><span><span class="co"># here just to show the functionality</span></span></span>
<span class="r-in"><span><span class="va">tse_temp</span> <span class="op">&lt;-</span> <span class="fu">mergeSEs</span><span class="op">(</span><span class="va">tse</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>, <span class="va">tse</span><span class="op">[</span><span class="fl">5</span><span class="op">:</span><span class="fl">100</span>, <span class="fl">11</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span>, join <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Merging with left join...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1/2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2/2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="va">tse_temp</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 10 20 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(10): Abiotrophia Achromobacter ... Azospirillum Bartonella</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(20): AQC1cm AQC4cm ... C D</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(16): X.SampleID Primer ... Age ClinicalStatus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># If your objects contain samples that describe one and same sample,</span></span></span>
<span class="r-in"><span><span class="co"># you can collapse equally named samples to one by specifying 'collapse.cols'</span></span></span>
<span class="r-in"><span><span class="va">tse_temp</span> <span class="op">&lt;-</span> <span class="fu">mergeSEs</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">tse</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">1</span><span class="op">]</span>, <span class="va">tse</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span>, <span class="fl">1</span><span class="op">]</span>, <span class="va">tse</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">]</span><span class="op">)</span>, </span></span>
<span class="r-in"><span>                       collapse.cols <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                       join <span class="op">=</span> <span class="st">"inner"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Merging with inner join...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1/3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2/3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3/3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="va">tse_temp</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 5 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(5): Abiotrophia Achromobacter Acidobacteria_Gp1_Gp1 Acidovorax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Aerococcus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(1): AM.AD.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(16): X.SampleID Primer ... Age ClinicalStatus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Merge all available assays</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="transformAssay.html">transformAssay</a></span><span class="op">(</span><span class="va">tse</span>, method<span class="op">=</span><span class="st">"relabundance"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">ts1</span> <span class="op">&lt;-</span> <span class="fu"><a href="transformAssay.html">transformAssay</a></span><span class="op">(</span><span class="va">tse1</span>, method<span class="op">=</span><span class="st">"relabundance"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tse_temp</span> <span class="op">&lt;-</span> <span class="fu">mergeSEs</span><span class="op">(</span><span class="va">tse</span>, <span class="va">tse1</span>, assay.type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assayNames</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following assay(s) was not found from all the objects so it is dropped from the output: 'relabundance'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Merging with full join...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1/2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2/2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Merging rowTree...</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>rowTree(s) does not match with the data so it is discarded.</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Felix G.M. Ernst, Sudarshan A. Shetty, Tuomas Borman, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer></div>






  </body></html>

